### PR TITLE
Support adding additional table headers 

### LINKED ISSUE #272: Error: The table already has a header, add additional rows to the existing table header instead
I have asked [this question](https://stackoverflow.com/questions/70349880/error-the-table-already-has-a-header-add-additional-rows-to-the-existing-tabl) on Stackoverflow but, with no answer after 24 hours, I have little hope to get one, now.

So, how do one "add additional rows to the existing table header" exactly since `doc.table().header()` already returns a row, and calling `doc.table().header()` multiple times throws this error?

### PR DESCRIPTION
**Pull request by KindaOK**
Resolves #272

Adds the option to create multiple table headers provided that no rows have been created yet.

Currently defers header rendering until the first row is placed. This does change existing behavior in that respect.

Passes all tests locally.

### CODE REVIEW

File: lib/table.js [hunk 1]
```diff
@@ -167,16 +173,20 @@ module.exports = class Table {
   }
 
   row(opts) {
+    // Defer rendering of the headers until a row is added
+    if (this._rowCount === this._headers.length) {
+      this._pending.push(() => this._renderHeader())
+    }
```
Comment (rkusa) [inline lib/table.js:179]: I don't mind this getting deferred. However, I think that we now need a unit test for a table that has a header, but no rows, and make sure the header is still rendered. Because this probably needs some additional code to work. Something like check in `_end()` whether the header got already rendered, and if not, execute `this._pending.push(() => this._renderHeader())` there.

**Review by rkusa — CHANGES_REQUESTED** [review]
Well done, kudos for digging through this rather old code 👍. I think there is only one thing left (see inline comment).

**Comment by KindaOK** [comment]
Fixed the issue. Also added some code to render the bottom border of a header along with unit tests.

**Comment by rkusa** [comment]
While preparing a release I've just noticed the following warnings in the `simple-multiple-headers` test:

```
# table/simple-multiple-headers
set cursor.y to negative value
set cursor.y to negative value
set cursor.y to negative value
set cursor.y to negative value
set cursor.y to negative value
set cursor.y to negative value
ok 116 simple-multiple-headers
```

Any idea what could cause that? If not, don't worry about it, I can also look into it

**Comment by KindaOK** [comment]
It has to do with the size of the headers and fitting them on a page. Some experimentation on the test is below.

| Num Headers (`i < x`) | Vertical Padding | Warning Messages |
| --- | --- | --- |
| 3 | 10 | None |
| 3 | 20 | 6 |
| 3 | 30 | 9 |
| 3 | 40 | **None** |
| 4 | 20 | 3 |
| 8 | 10 | **None** |
| 9 | 10 | 9 |
| 10 | 10 | **None** |


Error is reached here.
https://github.com/rkusa/pdfjs/blob/55a662f4bc40109d49ffee0b4bd3379bbf486006/lib/cell.js#L248

**Comment by rkusa** [comment]
I removed the warning altogether. It was also logged in other tests and there are cases where a negative y isn't an issue (as content is rendered to a "content chunk" that is later placed e.g. on the next page with another total vertical offset).
